pkg/monitoring/rules/recordingrules/operator.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2024 The KubeVirt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package recordingrules
+
+import (
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+	"github.com/machadovilaca/operator-observability/pkg/operatorrules"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	abnormalLabelKey = "reason"
+)
+
+var operatorRecordingRules = []operatorrules.RecordingRule{
+	{
+		MetricsOpts: operatormetrics.MetricOpts{
+			Name: "kubevirt_working_set_memory_exceeded",
+			Help: "The pod with the highest exceeded memory for each container based on the working set.",
+			ConstLabels: map[string]string{
+				abnormalLabelKey: "WorkingSetMemoryExceeded",
+			},
+		},
+		MetricType: operatormetrics.GaugeType,
+		Expr:       intstr.FromString("max by(container, namespace)(container_memory_working_set_bytes{container=~\"virt-controller|virt-api|virt-handler|virt-operator\"}  - on(pod) group_left(node) (kube_pod_container_resource_requests{container=~\"virt-controller|virt-api|virt-handler|virt-operator\",resource=\"memory\"}))"),
```

Comment:
please don't do that as Prometheus is going to pull all `container_memory_working_set_bytes` series which is suboptimal. Same goes wiht the `virt-.*` matcher: it may match user workload containers.

Reply:
adding the namespace filter as in https://github.com/kubevirt/kubevirt/pull/11557/files#r1535546967 would fix the issue you mentioned for the namespace,

Reply:
for the full series

even using p.e. `max by(container, namespace)(container_memory_working_set_bytes{container=~"virt-.*"} - on(node, namespace, pod, container) (kube_pod_container_resource_requests{resource="memory"}))`

would Prometheus optimize the query, and reduce the number of results from the second expression?

Reply:
changed to Joao's latest suggestion

Reply:
Virt-* will also include virt-launcher. We don't need that. Also we need to specifically state what pods we need track. It's not best practice to do it differently.

Reply:
> would Prometheus optimize the query, and reduce the number of results from the second expression?

no